### ISSUE TITLE
Testing the new compression feature with repo conversation: Works like a charm

### ISSUE THREAD

**Issue by JsBergbau** [restic/restic#3823]
After the compression performance test here https://github.com/restic/restic/issues/3821 I decided to convert a repo with about 2 TB of data and do some tests. TL;DR everything worked fine.

`restic version
restic 0.13.1 (v0.13.0-251-g98a3125c) compiled with go1.18.1 on windows/amd64`
crosscompiled with Ubuntu for Windows

Migrated via 
```
restic migrate  upgrade_repo_v2
resticcomp prune --repack-uncompressed --compression max --no-cache
```
Took about 30 hours from a NAS with AMD Ryzen 5 Pro 5650GE

2 TB Repo has about 140 snapshots covering about 1 year. 

After migration size was about 150 GB or 7,4 % reduced which I find impressive because a lot of uncompressible data like images, videos and documents are in that repo.

After conversion I restored a folder from a about 9 month old snapshot with about 200 GB in size. No errors were reported, I took a look into about 10 files. They were intact.

After that I restored another folder with about 60 GB in size. Also no errors were reported. There I've tested about 20 rar fieles > 300 MB and checked for errors. No errors were found.

After that I've added 3 snapshots. I've removed the first 2 snapshots and ran prune command.

After that I've restore 2 rar files with 5 GB and 3 GB from a 3 month old snapshot and tested the archive. Also no errors were found.

Finally I've restored a 12 GB folder from the first snapshot. Also no errors were reported and having a brief look into the audios and videos showed everything fine.

Hope this test on a real repo / data is helpful to you and hopefully a new release is soon made with the new compression feature. Its a really milestone for me and probably for a lot of other users.

PS:
I noticed restic taking 4.5 GB memory during repo conversation / compressing old packfiles of a 9 TB repo. That could lead to some problems on some machines, but probably restic takes so many memory because of the 16 threads and 16GB available memory on Ryzen 7 5700U. Did not have a look at memory usage on Ryzen 5 machine with 2 TB repo. For me memory usage is ok. I just wanted to note that

![grafik](https://user-images.githubusercontent.com/37013344/178849519-d8380b18-7895-4789-b120-c940c7601ba3.png)

**Comment by MichaelEischer** [comment]
Thanks for testing and nice to hear that everything worked fine.

> I noticed restic taking 4.5 GB memory during repo conversation / compressing old packfiles of a 9 TB repo.

How much memory does restic require when backing up to that repository? But as long as the memory usage is stable, it should be fine.

Besides that there shouldn't be anything to do here? So I'll close this issue in the next week or some time after the next release if I forget.

**Comment by JsBergbau** [comment]
> How much memory does restic require when backing up to that repository? But as long as the memory usage is stable, it should be fine.
I'll report when conversation is done. But it rarely can't be more than 1 GB because it is done on a Raspberry PI 400 with 4 GB of RAM where already 2 GB are used by other processes and I've never noticed any memory issues.

> Besides that there shouldn't be anything to do here? So I'll close this issue in the next week or some time after the next release if I forget.

Yes there is nothing to do from your side. I'll close with reporting the memory needed for backing up. Do you already have a date for next release with compression feature?

**Comment by MichaelEischer** [comment]
> Do you already have a date for next release with compression feature?

No, not yet. I still want to get a few PRs merged first, but it's not very far out anymore, I hope.

**Comment by JsBergbau** [comment]
Actually restic needs on the raspberry PI 400 with 4 GB of memory quite a lot memory to backup. 
2,78 GB of virtual memory and 1,4 GB while iterating the files
![grafik](https://user-images.githubusercontent.com/37013344/180082660-f50fc9c2-64a5-43cc-bec9-59eeaea00a58.png)

When getting to a file which is added to the repo memory usage goes up to 2 GB with 3,1 GB virtual. If seen the swap usage climbing a lot (max 8 GB)

![grafik](https://user-images.githubusercontent.com/37013344/180082829-13f85ae8-19ed-4e4b-b095-84df91c2f10f.png)

**Comment by JsBergbau** [comment]
Just want to add. I've also used the new RESTIC_PACK_SIZE feature. I've already converted all my repositories with RESTIC_COMPRESSION=max and packsizes of at least 32 MB up to 128 GB. In total it should be around 17 TB (in compressed form). 
I've also checked the biggest repositories with `restic check --read-data` and also the repositories that profitated most from compression. There was one repository that was smaller by a factor of almost 7 with compression. In total this should be about 14 - 15 TB of repository (in compressed state). All repositories passed the check. I mention this especially because the release notes state this feature is not tested very much and repositories should be checked more often https://github.com/restic/restic/blob/master/changelog/unreleased/issue-21
